Sources/InfluxDBSwift/FluxCSVParser.swift

```diff
@@ -123,41 +127,58 @@ internal class FluxCSVParser {
             let column: QueryAPI.FluxColumn = table.columns[$0.offset]
             column.name = $0.element
         }
+
+        let duplicates = Dictionary(grouping: table.columns.compactMap { $0.name }) { $0 }.filter { $1.count > 1 }.keys
+        if !duplicates.isEmpty {
+            print("""
+                  The response contains columns with duplicated names: \(duplicates.joined(separator: ", "))
+                  You should use the 'FluxRecord.row' to access your data instead of 'FluxRecord.values' dictionary.
+                  """)
+        }
     }
 
     private func parseRow(row: [String]) throws -> QueryAPI.FluxRecord {
         guard let table = table else {
             throw InfluxDBClient.InfluxDBError.generic(Self.errorMessage)
         }
-
+        var recordRow: [Any] = Array()
```

Comment:
Following code is little bit complicated... What do you think about something like?

```swift
var recordRow: [Any] = Array()
let values: [String: Decodable] = table.columns.reduce(into: [String: Decodable]()) { result, column in
    var value: String = row[column.index + 1]
    if value.isEmpty {
        value = column.defaultValue
    }
    switch column.dataType {
    case "boolean":
        result[column.name] = (value as NSString).boolValue
    case "unsignedLong":
        result[column.name] = UInt64(value)
    case "long":
        result[column.name] = Int64(value)
    case "double":
        result[column.name] = Double(value)
    case "string":
        result[column.name] = value
    case "dateTime:RFC3339", "dateTime:RFC3339Nano":
        result[column.name] = CodableHelper.dateFormatter.date(from: value)
    case "base64Binary":
        result[column.name] = Data(base64Encoded: value)
    case "duration":
        result[column.name] = Int64(value)
    default:
        result[column.name] = value
    }
    recordRow.append(result[column.name] as Any)
}
```